{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## KEGG Pathways Gene Selection"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Connecting to KEGG Pathways Database with BioPython "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'path:hsa00010\\tGlycolysis / Gluconeogenesis - Homo sapiens (human)\\npath:hsa00020\\tCitrate cycle (TCA cycle) - Homo sapiens (human)\\npath:hsa00030\\tPentose phosphate pathway - Homo sapiens (human)\\npath:hsa00040\\tPentose and glucuronate interconversions - Homo sapiens (human)\\npath:hsa00051\\tFructose and mannose metabolism - Homo sapiens (human)\\npath:hsa00052\\tGalactose metabolism - Homo sapiens (human)\\npath:hsa00053\\tAscorbate and aldarate metabolism - Homo sapiens (human)\\npath:hsa00061\\tFatty acid biosynthesis - Homo sapiens (human)\\npath:hsa00062\\tFatty acid elongation - Homo sapiens (human)\\npath:hsa00071\\tFatty acid degradation - Homo sapiens (human)\\npath:hsa00072\\tSynthesis and degradation of ketone bodies - Homo sapiens (human)\\npath:hsa00100\\tSteroid biosynthesis - Homo sapiens (human)\\npath:hsa00120\\tPrimary bile acid biosynthesis - Homo sapiens (human)\\npath:hsa00130\\tUbiquinone and other terpenoid-quinone biosynthesis - Homo sapiens (human)\\npath:hsa00140\\tSteroid hormone biosynthesis - Homo sapiens (human)\\npath:hsa00190\\tOxidative phosphorylation - Homo sapiens (human)\\npath:hsa00220\\tArginine biosynthesis - Homo sapiens (human)\\npath:hsa00230\\tPurine metabolism - Homo sapiens (human)\\npath:hsa00232\\tCaffeine metabolism - Homo sapiens (human)\\npath:hsa00240\\tPyrimidine metabolism - Homo sapiens (human)\\npath:hsa00250\\tAlanine, aspartate and glutamate metabolism - Homo sapiens (human)\\npath:hsa00260\\tGlycine, serine and threonine metabolism - Homo sapiens (human)\\npath:hsa00270\\tCysteine and methionine metabolism - Homo sapiens (human)\\npath:hsa00280\\tValine, leucine and isoleucine degradation - Homo sapiens (human)\\npath:hsa00290\\tValine, leucine and isoleucine biosynthesis - Homo sapiens (human)\\npath:hsa00310\\tLysine degradation - Homo sapiens (human)\\npath:hsa00330\\tArginine and proline metabolism - Homo sapiens (human)\\npath:hsa00340\\tHistidine metabolism - Homo sapiens (human)\\npath:hsa00350\\tTyrosine metabolism - Homo sapiens (human)\\npath:hsa00360\\tPhenylalanine metabolism - Homo sapiens (human)\\npath:hsa00380\\tTryptophan metabolism - Homo sapiens (human)\\npath:hsa00400\\tPhenylalanine, tyrosine and tryptophan biosynthesis - Homo sapiens (human)\\npath:hsa00410\\tbeta-Alanine metabolism - Homo sapiens (human)\\npath:hsa00430\\tTaurine and hypotaurine metabolism - Homo sapiens (human)\\npath:hsa00440\\tPhosphonate and phosphinate metabolism - Homo sapiens (human)\\npath:hsa00450\\tSelenocompound metabolism - Homo sapiens (human)\\npath:hsa00471\\tD-Glutamine and D-glutamate metabolism - Homo sapiens (human)\\npath:hsa00472\\tD-Arginine and D-ornithine metabolism - Homo sapiens (human)\\npath:hsa00480\\tGlutathione metabolism - Homo sapiens (human)\\npath:hsa00500\\tStarch and sucrose metabolism - Homo sapiens (human)\\npath:hsa00510\\tN-Glycan biosynthesis - Homo sapiens (human)\\npath:hsa00511\\tOther glycan degradation - Homo sapiens (human)\\npath:hsa00512\\tMucin type O-glycan biosynthesis - Homo sapiens (human)\\npath:hsa00514\\tOther types of O-glycan biosynthesis - Homo sapiens (human)\\npath:hsa00515\\tMannose type O-glycan biosynthesis - Homo sapiens (human)\\npath:hsa00520\\tAmino sugar and nucleotide sugar metabolism - Homo sapiens (human)\\npath:hsa00524\\tNeomycin, kanamycin and gentamicin biosynthesis - Homo sapiens (human)\\npath:hsa00531\\tGlycosaminoglycan degradation - Homo sapiens (human)\\npath:hsa00532\\tGlycosaminoglycan biosynthesis - chondroitin sulfate / dermatan sulfate - Homo sapiens (human)\\npath:hsa00533\\tGlycosaminoglycan biosynthesis - keratan sulfate - Homo sapiens (human)\\npath:hsa00534\\tGlycosaminoglycan biosynthesis - heparan sulfate / heparin - Homo sapiens (human)\\npath:hsa00561\\tGlycerolipid metabolism - Homo sapiens (human)\\npath:hsa00562\\tInositol phosphate metabolism - Homo sapiens (human)\\npath:hsa00563\\tGlycosylphosphatidylinositol (GPI)-anchor biosynthesis - Homo sapiens (human)\\npath:hsa00564\\tGlycerophospholipid metabolism - Homo sapiens (human)\\npath:hsa00565\\tEther lipid metabolism - Homo sapiens (human)\\npath:hsa00590\\tArachidonic acid metabolism - Homo sapiens (human)\\npath:hsa00591\\tLinoleic acid metabolism - Homo sapiens (human)\\npath:hsa00592\\talpha-Linolenic acid metabolism - Homo sapiens (human)\\npath:hsa00600\\tSphingolipid metabolism - Homo sapiens (human)\\npath:hsa00601\\tGlycosphingolipid biosynthesis - lacto and neolacto series - Homo sapiens (human)\\npath:hsa00603\\tGlycosphingolipid biosynthesis - globo and isoglobo series - Homo sapiens (human)\\npath:hsa00604\\tGlycosphingolipid biosynthesis - ganglio series - Homo sapiens (human)\\npath:hsa00620\\tPyruvate metabolism - Homo sapiens (human)\\npath:hsa00630\\tGlyoxylate and dicarboxylate metabolism - Homo sapiens (human)\\npath:hsa00640\\tPropanoate metabolism - Homo sapiens (human)\\npath:hsa00650\\tButanoate metabolism - Homo sapiens (human)\\npath:hsa00670\\tOne carbon pool by folate - Homo sapiens (human)\\npath:hsa00730\\tThiamine metabolism - Homo sapiens (human)\\npath:hsa00740\\tRiboflavin metabolism - Homo sapiens (human)\\npath:hsa00750\\tVitamin B6 metabolism - Homo sapiens (human)\\npath:hsa00760\\tNicotinate and nicotinamide metabolism - Homo sapiens (human)\\npath:hsa00770\\tPantothenate and CoA biosynthesis - Homo sapiens (human)\\npath:hsa00780\\tBiotin metabolism - Homo sapiens (human)\\npath:hsa00785\\tLipoic acid metabolism - Homo sapiens (human)\\npath:hsa00790\\tFolate biosynthesis - Homo sapiens (human)\\npath:hsa00830\\tRetinol metabolism - Homo sapiens (human)\\npath:hsa00860\\tPorphyrin and chlorophyll metabolism - Homo sapiens (human)\\npath:hsa00900\\tTerpenoid backbone biosynthesis - Homo sapiens (human)\\npath:hsa00910\\tNitrogen metabolism - Homo sapiens (human)\\npath:hsa00920\\tSulfur metabolism - Homo sapiens (human)\\npath:hsa00970\\tAminoacyl-tRNA biosynthesis - Homo sapiens (human)\\npath:hsa00980\\tMetabolism of xenobiotics by cytochrome P450 - Homo sapiens (human)\\npath:hsa00982\\tDrug metabolism - cytochrome P450 - Homo sapiens (human)\\npath:hsa00983\\tDrug metabolism - other enzymes - Homo sapiens (human)\\npath:hsa01040\\tBiosynthesis of unsaturated fatty acids - Homo sapiens (human)\\npath:hsa01100\\tMetabolic pathways - Homo sapiens (human)\\npath:hsa01200\\tCarbon metabolism - Homo sapiens (human)\\npath:hsa01210\\t2-Oxocarboxylic acid metabolism - Homo sapiens (human)\\npath:hsa01212\\tFatty acid metabolism - Homo sapiens (human)\\npath:hsa01230\\tBiosynthesis of amino acids - Homo sapiens (human)\\npath:hsa01521\\tEGFR tyrosine kinase inhibitor resistance - Homo sapiens (human)\\npath:hsa01522\\tEndocrine resistance - Homo sapiens (human)\\npath:hsa01523\\tAntifolate resistance - Homo sapiens (human)\\npath:hsa01524\\tPlatinum drug resistance - Homo sapiens (human)\\npath:hsa02010\\tABC transporters - Homo sapiens (human)\\npath:hsa03008\\tRibosome biogenesis in eukaryotes - Homo sapiens (human)\\npath:hsa03010\\tRibosome - Homo sapiens (human)\\npath:hsa03013\\tRNA transport - Homo sapiens (human)\\npath:hsa03015\\tmRNA surveillance pathway - Homo sapiens (human)\\npath:hsa03018\\tRNA degradation - Homo sapiens (human)\\npath:hsa03020\\tRNA polymerase - Homo sapiens (human)\\npath:hsa03022\\tBasal transcription factors - Homo sapiens (human)\\npath:hsa03030\\tDNA replication - Homo sapiens (human)\\npath:hsa03040\\tSpliceosome - Homo sapiens (human)\\npath:hsa03050\\tProteasome - Homo sapiens (human)\\npath:hsa03060\\tProtein export - Homo sapiens (human)\\npath:hsa03320\\tPPAR signaling pathway - Homo sapiens (human)\\npath:hsa03410\\tBase excision repair - Homo sapiens (human)\\npath:hsa03420\\tNucleotide excision repair - Homo sapiens (human)\\npath:hsa03430\\tMismatch repair - Homo sapiens (human)\\npath:hsa03440\\tHomologous recombination - Homo sapiens (human)\\npath:hsa03450\\tNon-homologous end-joining - Homo sapiens (human)\\npath:hsa03460\\tFanconi anemia pathway - Homo sapiens (human)\\npath:hsa04010\\tMAPK signaling pathway - Homo sapiens (human)\\npath:hsa04012\\tErbB signaling pathway - Homo sapiens (human)\\npath:hsa04014\\tRas signaling pathway - Homo sapiens (human)\\npath:hsa04015\\tRap1 signaling pathway - Homo sapiens (human)\\npath:hsa04020\\tCalcium signaling pathway - Homo sapiens (human)\\npath:hsa04022\\tcGMP-PKG signaling pathway - Homo sapiens (human)\\npath:hsa04024\\tcAMP signaling pathway - Homo sapiens (human)\\npath:hsa04060\\tCytokine-cytokine receptor interaction - Homo sapiens (human)\\npath:hsa04062\\tChemokine signaling pathway - Homo sapiens (human)\\npath:hsa04064\\tNF-kappa B signaling pathway - Homo sapiens (human)\\npath:hsa04066\\tHIF-1 signaling pathway - Homo sapiens (human)\\npath:hsa04068\\tFoxO signaling pathway - Homo sapiens (human)\\npath:hsa04070\\tPhosphatidylinositol signaling system - Homo sapiens (human)\\npath:hsa04071\\tSphingolipid signaling pathway - Homo sapiens (human)\\npath:hsa04072\\tPhospholipase D signaling pathway - Homo sapiens (human)\\npath:hsa04080\\tNeuroactive ligand-receptor interaction - Homo sapiens (human)\\npath:hsa04110\\tCell cycle - Homo sapiens (human)\\npath:hsa04114\\tOocyte meiosis - Homo sapiens (human)\\npath:hsa04115\\tp53 signaling pathway - Homo sapiens (human)\\npath:hsa04120\\tUbiquitin mediated proteolysis - Homo sapiens (human)\\npath:hsa04122\\tSulfur relay system - Homo sapiens (human)\\npath:hsa04130\\tSNARE interactions in vesicular transport - Homo sapiens (human)\\npath:hsa04136\\tAutophagy - other - Homo sapiens (human)\\npath:hsa04137\\tMitophagy - animal - Homo sapiens (human)\\npath:hsa04140\\tAutophagy - animal - Homo sapiens (human)\\npath:hsa04141\\tProtein processing in endoplasmic reticulum - Homo sapiens (human)\\npath:hsa04142\\tLysosome - Homo sapiens (human)\\npath:hsa04144\\tEndocytosis - Homo sapiens (human)\\npath:hsa04145\\tPhagosome - Homo sapiens (human)\\npath:hsa04146\\tPeroxisome - Homo sapiens (human)\\npath:hsa04150\\tmTOR signaling pathway - Homo sapiens (human)\\npath:hsa04151\\tPI3K-Akt signaling pathway - Homo sapiens (human)\\npath:hsa04152\\tAMPK signaling pathway - Homo sapiens (human)\\npath:hsa04210\\tApoptosis - Homo sapiens (human)\\npath:hsa04211\\tLongevity regulating pathway - Homo sapiens (human)\\npath:hsa04213\\tLongevity regulating pathway - multiple species - Homo sapiens (human)\\npath:hsa04215\\tApoptosis - multiple species - Homo sapiens (human)\\npath:hsa04216\\tFerroptosis - Homo sapiens (human)\\npath:hsa04217\\tNecroptosis - Homo sapiens (human)\\npath:hsa04218\\tCellular senescence - Homo sapiens (human)\\npath:hsa04260\\tCardiac muscle contraction - Homo sapiens (human)\\npath:hsa04261\\tAdrenergic signaling in cardiomyocytes - Homo sapiens (human)\\npath:hsa04270\\tVascular smooth muscle contraction - Homo sapiens (human)\\npath:hsa04310\\tWnt signaling pathway - Homo sapiens (human)\\npath:hsa04330\\tNotch signaling pathway - Homo sapiens (human)\\npath:hsa04340\\tHedgehog signaling pathway - Homo sapiens (human)\\npath:hsa04350\\tTGF-beta signaling pathway - Homo sapiens (human)\\npath:hsa04360\\tAxon guidance - Homo sapiens (human)\\npath:hsa04370\\tVEGF signaling pathway - Homo sapiens (human)\\npath:hsa04371\\tApelin signaling pathway - Homo sapiens (human)\\npath:hsa04380\\tOsteoclast differentiation - Homo sapiens (human)\\npath:hsa04390\\tHippo signaling pathway - Homo sapiens (human)\\npath:hsa04392\\tHippo signaling pathway - multiple species - Homo sapiens (human)\\npath:hsa04510\\tFocal adhesion - Homo sapiens (human)\\npath:hsa04512\\tECM-receptor interaction - Homo sapiens (human)\\npath:hsa04514\\tCell adhesion molecules (CAMs) - Homo sapiens (human)\\npath:hsa04520\\tAdherens junction - Homo sapiens (human)\\npath:hsa04530\\tTight junction - Homo sapiens (human)\\npath:hsa04540\\tGap junction - Homo sapiens (human)\\npath:hsa04550\\tSignaling pathways regulating pluripotency of stem cells - Homo sapiens (human)\\npath:hsa04610\\tComplement and coagulation cascades - Homo sapiens (human)\\npath:hsa04611\\tPlatelet activation - Homo sapiens (human)\\npath:hsa04612\\tAntigen processing and presentation - Homo sapiens (human)\\npath:hsa04614\\tRenin-angiotensin system - Homo sapiens (human)\\npath:hsa04620\\tToll-like receptor signaling pathway - Homo sapiens (human)\\npath:hsa04621\\tNOD-like receptor signaling pathway - Homo sapiens (human)\\npath:hsa04622\\tRIG-I-like receptor signaling pathway - Homo sapiens (human)\\npath:hsa04623\\tCytosolic DNA-sensing pathway - Homo sapiens (human)\\npath:hsa04625\\tC-type lectin receptor signaling pathway - Homo sapiens (human)\\npath:hsa04630\\tJAK-STAT signaling pathway - Homo sapiens (human)\\npath:hsa04640\\tHematopoietic cell lineage - Homo sapiens (human)\\npath:hsa04650\\tNatural killer cell mediated cytotoxicity - Homo sapiens (human)\\npath:hsa04657\\tIL-17 signaling pathway - Homo sapiens (human)\\npath:hsa04658\\tTh1 and Th2 cell differentiation - Homo sapiens (human)\\npath:hsa04659\\tTh17 cell differentiation - Homo sapiens (human)\\npath:hsa04660\\tT cell receptor signaling pathway - Homo sapiens (human)\\npath:hsa04662\\tB cell receptor signaling pathway - Homo sapiens (human)\\npath:hsa04664\\tFc epsilon RI signaling pathway - Homo sapiens (human)\\npath:hsa04666\\tFc gamma R-mediated phagocytosis - Homo sapiens (human)\\npath:hsa04668\\tTNF signaling pathway - Homo sapiens (human)\\npath:hsa04670\\tLeukocyte transendothelial migration - Homo sapiens (human)\\npath:hsa04672\\tIntestinal immune network for IgA production - Homo sapiens (human)\\npath:hsa04710\\tCircadian rhythm - Homo sapiens (human)\\npath:hsa04713\\tCircadian entrainment - Homo sapiens (human)\\npath:hsa04714\\tThermogenesis - Homo sapiens (human)\\npath:hsa04720\\tLong-term potentiation - Homo sapiens (human)\\npath:hsa04721\\tSynaptic vesicle cycle - Homo sapiens (human)\\npath:hsa04722\\tNeurotrophin signaling pathway - Homo sapiens (human)\\npath:hsa04723\\tRetrograde endocannabinoid signaling - Homo sapiens (human)\\npath:hsa04724\\tGlutamatergic synapse - Homo sapiens (human)\\npath:hsa04725\\tCholinergic synapse - Homo sapiens (human)\\npath:hsa04726\\tSerotonergic synapse - Homo sapiens (human)\\npath:hsa04727\\tGABAergic synapse - Homo sapiens (human)\\npath:hsa04728\\tDopaminergic synapse - Homo sapiens (human)\\npath:hsa04730\\tLong-term depression - Homo sapiens (human)\\npath:hsa04740\\tOlfactory transduction - Homo sapiens (human)\\npath:hsa04742\\tTaste transduction - Homo sapiens (human)\\npath:hsa04744\\tPhototransduction - Homo sapiens (human)\\npath:hsa04750\\tInflammatory mediator regulation of TRP channels - Homo sapiens (human)\\npath:hsa04810\\tRegulation of actin cytoskeleton - Homo sapiens (human)\\npath:hsa04910\\tInsulin signaling pathway - Homo sapiens (human)\\npath:hsa04911\\tInsulin secretion - Homo sapiens (human)\\npath:hsa04912\\tGnRH signaling pathway - Homo sapiens (human)\\npath:hsa04913\\tOvarian steroidogenesis - Homo sapiens (human)\\npath:hsa04914\\tProgesterone-mediated oocyte maturation - Homo sapiens (human)\\npath:hsa04915\\tEstrogen signaling pathway - Homo sapiens (human)\\npath:hsa04916\\tMelanogenesis - Homo sapiens (human)\\npath:hsa04917\\tProlactin signaling pathway - Homo sapiens (human)\\npath:hsa04918\\tThyroid hormone synthesis - Homo sapiens (human)\\npath:hsa04919\\tThyroid hormone signaling pathway - Homo sapiens (human)\\npath:hsa04920\\tAdipocytokine signaling pathway - Homo sapiens (human)\\npath:hsa04921\\tOxytocin signaling pathway - Homo sapiens (human)\\npath:hsa04922\\tGlucagon signaling pathway - Homo sapiens (human)\\npath:hsa04923\\tRegulation of lipolysis in adipocytes - Homo sapiens (human)\\npath:hsa04924\\tRenin secretion - Homo sapiens (human)\\npath:hsa04925\\tAldosterone synthesis and secretion - Homo sapiens (human)\\npath:hsa04926\\tRelaxin signaling pathway - Homo sapiens (human)\\npath:hsa04927\\tCortisol synthesis and secretion - Homo sapiens (human)\\npath:hsa04928\\tParathyroid hormone synthesis, secretion and action - Homo sapiens (human)\\npath:hsa04930\\tType II diabetes mellitus - Homo sapiens (human)\\npath:hsa04931\\tInsulin resistance - Homo sapiens (human)\\npath:hsa04932\\tNon-alcoholic fatty liver disease (NAFLD) - Homo sapiens (human)\\npath:hsa04933\\tAGE-RAGE signaling pathway in diabetic complications - Homo sapiens (human)\\npath:hsa04934\\tCushing syndrome - Homo sapiens (human)\\npath:hsa04940\\tType I diabetes mellitus - Homo sapiens (human)\\npath:hsa04950\\tMaturity onset diabetes of the young - Homo sapiens (human)\\npath:hsa04960\\tAldosterone-regulated sodium reabsorption - Homo sapiens (human)\\npath:hsa04961\\tEndocrine and other factor-regulated calcium reabsorption - Homo sapiens (human)\\npath:hsa04962\\tVasopressin-regulated water reabsorption - Homo sapiens (human)\\npath:hsa04964\\tProximal tubule bicarbonate reclamation - Homo sapiens (human)\\npath:hsa04966\\tCollecting duct acid secretion - Homo sapiens (human)\\npath:hsa04970\\tSalivary secretion - Homo sapiens (human)\\npath:hsa04971\\tGastric acid secretion - Homo sapiens (human)\\npath:hsa04972\\tPancreatic secretion - Homo sapiens (human)\\npath:hsa04973\\tCarbohydrate digestion and absorption - Homo sapiens (human)\\npath:hsa04974\\tProtein digestion and absorption - Homo sapiens (human)\\npath:hsa04975\\tFat digestion and absorption - Homo sapiens (human)\\npath:hsa04976\\tBile secretion - Homo sapiens (human)\\npath:hsa04977\\tVitamin digestion and absorption - Homo sapiens (human)\\npath:hsa04978\\tMineral absorption - Homo sapiens (human)\\npath:hsa04979\\tCholesterol metabolism - Homo sapiens (human)\\npath:hsa05010\\tAlzheimer disease - Homo sapiens (human)\\npath:hsa05012\\tParkinson disease - Homo sapiens (human)\\npath:hsa05014\\tAmyotrophic lateral sclerosis (ALS) - Homo sapiens (human)\\npath:hsa05016\\tHuntington disease - Homo sapiens (human)\\npath:hsa05020\\tPrion diseases - Homo sapiens (human)\\npath:hsa05030\\tCocaine addiction - Homo sapiens (human)\\npath:hsa05031\\tAmphetamine addiction - Homo sapiens (human)\\npath:hsa05032\\tMorphine addiction - Homo sapiens (human)\\npath:hsa05033\\tNicotine addiction - Homo sapiens (human)\\npath:hsa05034\\tAlcoholism - Homo sapiens (human)\\npath:hsa05100\\tBacterial invasion of epithelial cells - Homo sapiens (human)\\npath:hsa05110\\tVibrio cholerae infection - Homo sapiens (human)\\npath:hsa05120\\tEpithelial cell signaling in Helicobacter pylori infection - Homo sapiens (human)\\npath:hsa05130\\tPathogenic Escherichia coli infection - Homo sapiens (human)\\npath:hsa05131\\tShigellosis - Homo sapiens (human)\\npath:hsa05132\\tSalmonella infection - Homo sapiens (human)\\npath:hsa05133\\tPertussis - Homo sapiens (human)\\npath:hsa05134\\tLegionellosis - Homo sapiens (human)\\npath:hsa05140\\tLeishmaniasis - Homo sapiens (human)\\npath:hsa05142\\tChagas disease (American trypanosomiasis) - Homo sapiens (human)\\npath:hsa05143\\tAfrican trypanosomiasis - Homo sapiens (human)\\npath:hsa05144\\tMalaria - Homo sapiens (human)\\npath:hsa05145\\tToxoplasmosis - Homo sapiens (human)\\npath:hsa05146\\tAmoebiasis - Homo sapiens (human)\\npath:hsa05150\\tStaphylococcus aureus infection - Homo sapiens (human)\\npath:hsa05152\\tTuberculosis - Homo sapiens (human)\\npath:hsa05160\\tHepatitis C - Homo sapiens (human)\\npath:hsa05161\\tHepatitis B - Homo sapiens (human)\\npath:hsa05162\\tMeasles - Homo sapiens (human)\\npath:hsa05163\\tHuman cytomegalovirus infection - Homo sapiens (human)\\npath:hsa05164\\tInfluenza A - Homo sapiens (human)\\npath:hsa05165\\tHuman papillomavirus infection - Homo sapiens (human)\\npath:hsa05166\\tHuman T-cell leukemia virus 1 infection - Homo sapiens (human)\\npath:hsa05167\\tKaposi sarcoma-associated herpesvirus infection - Homo sapiens (human)\\npath:hsa05168\\tHerpes simplex virus 1 infection - Homo sapiens (human)\\npath:hsa05169\\tEpstein-Barr virus infection - Homo sapiens (human)\\npath:hsa05170\\tHuman immunodeficiency virus 1 infection - Homo sapiens (human)\\npath:hsa05200\\tPathways in cancer - Homo sapiens (human)\\npath:hsa05202\\tTranscriptional misregulation in cancer - Homo sapiens (human)\\npath:hsa05203\\tViral carcinogenesis - Homo sapiens (human)\\npath:hsa05204\\tChemical carcinogenesis - Homo sapiens (human)\\npath:hsa05205\\tProteoglycans in cancer - Homo sapiens (human)\\npath:hsa05206\\tMicroRNAs in cancer - Homo sapiens (human)\\npath:hsa05210\\tColorectal cancer - Homo sapiens (human)\\npath:hsa05211\\tRenal cell carcinoma - Homo sapiens (human)\\npath:hsa05212\\tPancreatic cancer - Homo sapiens (human)\\npath:hsa05213\\tEndometrial cancer - Homo sapiens (human)\\npath:hsa05214\\tGlioma - Homo sapiens (human)\\npath:hsa05215\\tProstate cancer - Homo sapiens (human)\\npath:hsa05216\\tThyroid cancer - Homo sapiens (human)\\npath:hsa05217\\tBasal cell carcinoma - Homo sapiens (human)\\npath:hsa05218\\tMelanoma - Homo sapiens (human)\\npath:hsa05219\\tBladder cancer - Homo sapiens (human)\\npath:hsa05220\\tChronic myeloid leukemia - Homo sapiens (human)\\npath:hsa05221\\tAcute myeloid leukemia - Homo sapiens (human)\\npath:hsa05222\\tSmall cell lung cancer - Homo sapiens (human)\\npath:hsa05223\\tNon-small cell lung cancer - Homo sapiens (human)\\npath:hsa05224\\tBreast cancer - Homo sapiens (human)\\npath:hsa05225\\tHepatocellular carcinoma - Homo sapiens (human)\\npath:hsa05226\\tGastric cancer - Homo sapiens (human)\\npath:hsa05230\\tCentral carbon metabolism in cancer - Homo sapiens (human)\\npath:hsa05231\\tCholine metabolism in cancer - Homo sapiens (human)\\npath:hsa05310\\tAsthma - Homo sapiens (human)\\npath:hsa05320\\tAutoimmune thyroid disease - Homo sapiens (human)\\npath:hsa05321\\tInflammatory bowel disease (IBD) - Homo sapiens (human)\\npath:hsa05322\\tSystemic lupus erythematosus - Homo sapiens (human)\\npath:hsa05323\\tRheumatoid arthritis - Homo sapiens (human)\\npath:hsa05330\\tAllograft rejection - Homo sapiens (human)\\npath:hsa05332\\tGraft-versus-host disease - Homo sapiens (human)\\npath:hsa05340\\tPrimary immunodeficiency - Homo sapiens (human)\\npath:hsa05410\\tHypertrophic cardiomyopathy (HCM) - Homo sapiens (human)\\npath:hsa05412\\tArrhythmogenic right ventricular cardiomyopathy (ARVC) - Homo sapiens (human)\\npath:hsa05414\\tDilated cardiomyopathy (DCM) - Homo sapiens (human)\\npath:hsa05416\\tViral myocarditis - Homo sapiens (human)\\npath:hsa05418\\tFluid shear stress and atherosclerosis - Homo sapiens (human)\\n'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from Bio.KEGG import REST\n",
    "import pandas as pd\n",
    "human_pathways = REST.kegg_list(\"pathway\", \"hsa\").read()\n",
    "human_pathways"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Following code retrieves the HSA Pathway name from the above text output and stores in a list, \"Pathways\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filter all human pathways for repair pathways\n",
    "pathways = []\n",
    "for line in human_pathways.rstrip().split(\"\\n\"):\n",
    "    entry, description = line.split(\"\\t\")\n",
    " #   if \"repair\" in description:\n",
    "    pathways.append(entry)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Creates an empty dataframe with rows as pathways. Pathways will be filled in following steps with genes present from our Gene Expression matrix that are active in each pathway."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pathways</th>\n",
       "      <th>gene1</th>\n",
       "      <th>gene2</th>\n",
       "      <th>gene3</th>\n",
       "      <th>gene4</th>\n",
       "      <th>gene5</th>\n",
       "      <th>gene6</th>\n",
       "      <th>gene7</th>\n",
       "      <th>gene8</th>\n",
       "      <th>gene9</th>\n",
       "      <th>...</th>\n",
       "      <th>gene991</th>\n",
       "      <th>gene992</th>\n",
       "      <th>gene993</th>\n",
       "      <th>gene994</th>\n",
       "      <th>gene995</th>\n",
       "      <th>gene996</th>\n",
       "      <th>gene997</th>\n",
       "      <th>gene998</th>\n",
       "      <th>gene999</th>\n",
       "      <th>gene1000</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>path:hsa00010</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 1001 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        pathways  gene1  gene2  gene3  gene4  gene5  gene6  gene7  gene8  \\\n",
       "0  path:hsa00010      0      0      0      0      0      0      0      0   \n",
       "\n",
       "   gene9    ...     gene991  gene992  gene993  gene994  gene995  gene996  \\\n",
       "0      0    ...           0        0        0        0        0        0   \n",
       "\n",
       "   gene997  gene998  gene999  gene1000  \n",
       "0        0        0        0         0  \n",
       "\n",
       "[1 rows x 1001 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame()\n",
    "se = pd.Series(pathways)\n",
    "df['pathways']=se.values\n",
    "#df= df.transpose()\n",
    "list_name = []\n",
    "for i in range(1000):\n",
    "    list_name.append('gene'+ str(i + 1))\n",
    "list_name\n",
    "\n",
    "#df['genes']=list\n",
    "for gene in list_name:\n",
    "    df[gene] = 0\n",
    "df.head(1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Creates a function to fill each pathway with active genes by HGNC Symbol ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "def fill_gene(df, genes_total, start):\n",
    "    for j in range(len(genes_total)):\n",
    "        for i in range(len(genes_total[j])):\n",
    "            df.iloc[j + start,i+1] = genes_total[j][i]\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pathways</th>\n",
       "      <th>gene1</th>\n",
       "      <th>gene2</th>\n",
       "      <th>gene3</th>\n",
       "      <th>gene4</th>\n",
       "      <th>gene5</th>\n",
       "      <th>gene6</th>\n",
       "      <th>gene7</th>\n",
       "      <th>gene8</th>\n",
       "      <th>gene9</th>\n",
       "      <th>...</th>\n",
       "      <th>gene991</th>\n",
       "      <th>gene992</th>\n",
       "      <th>gene993</th>\n",
       "      <th>gene994</th>\n",
       "      <th>gene995</th>\n",
       "      <th>gene996</th>\n",
       "      <th>gene997</th>\n",
       "      <th>gene998</th>\n",
       "      <th>gene999</th>\n",
       "      <th>gene1000</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>path:hsa00010</td>\n",
       "      <td>3101</td>\n",
       "      <td>3098</td>\n",
       "      <td>3099</td>\n",
       "      <td>80201</td>\n",
       "      <td>2645</td>\n",
       "      <td>2821</td>\n",
       "      <td>5213</td>\n",
       "      <td>5214</td>\n",
       "      <td>5211</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 1001 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        pathways gene1 gene2 gene3  gene4 gene5 gene6 gene7 gene8 gene9  \\\n",
       "0  path:hsa00010  3101  3098  3099  80201  2645  2821  5213  5214  5211   \n",
       "\n",
       "    ...    gene991 gene992 gene993 gene994 gene995 gene996 gene997 gene998  \\\n",
       "0   ...          0       0       0       0       0       0       0       0   \n",
       "\n",
       "  gene999 gene1000  \n",
       "0       0        0  \n",
       "\n",
       "[1 rows x 1001 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get the genes for pathways and add them to a list\n",
    "genes_total = []\n",
    "for i in range(10):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df= fill_gene(df, genes_total, 0)\n",
    "df.head(1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The following code fills all pathways with their respective genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(10,17):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(18,25):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 18)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(25,26):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 25)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(27,37):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 27)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(37,47):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 37)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(47,60):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 47)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(60,65):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 60)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(65,68):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 65)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(69,80):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 69)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(80,82):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 80)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(84,90):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 84)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(90,96):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 90)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(97,107):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 97)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(107,117):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 107)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(117,130):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 117)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(130,140):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 130)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(140,144):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 140)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(145,150):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 145)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(150,160):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 150)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(160,164):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 160)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(165,169):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 165)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(170,180):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 170)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(180,190):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 180)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(190,195):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 190)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(196,200):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 196)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(200,205):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 200)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(206,209):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 206)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(210,215):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 210)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(216,219):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 216)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(220,230):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 220)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(230,240):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 230)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(240,250):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 240)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(250,260):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 250)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(260,270):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 260)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(270,280):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 270)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(280,290):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 280)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(290,300):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 290)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(300,310):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 300)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(310,320):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 310)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(320,330):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 320)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes_total = []\n",
    "for i in range(86,95):\n",
    "    genes = []\n",
    "    pathway_file = REST.kegg_get(pathways[i]).read()  # query and read each pathway\n",
    "    # iterate through each KEGG pathway file, keeping track of which section\n",
    "    # of the file we're in, only read the gene in each pathway\n",
    "    current_section = None\n",
    "    for line in pathway_file.rstrip().split(\"\\n\"):\n",
    "        section = line[:12].strip()  # section names are within 12 columns\n",
    "        if not section == \"\":\n",
    "            current_section = section\n",
    "\n",
    "        if current_section == \"GENE\":\n",
    "            gene_identifiers = line[12:].split(\"; \")[0]\n",
    "            gene_id, gene_symbol = gene_identifiers.split()\n",
    "\n",
    "            if not gene_symbol in genes:\n",
    "                genes.append(gene_id)\n",
    "    genes_total.append(genes)\n",
    "\n",
    "df = fill_gene(df, genes_total, 86)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv('df-geneid1.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
